pyccel/codegen/printing/fcode.py

```diff
@@ -285,7 +322,13 @@ def print_kind(self, expr):
         """
         Prints the kind(precision) of a literal value
         """
-        return iso_c_binding[self._print(expr.dtype)][expr.precision]
+        if self._print(expr.dtype) == 'complex':
+            return iso_c_binding[self._print(expr.dtype)][expr.precision]
+        index = iso_c_binding_check_index[self._print(expr.dtype)][expr.precision]
+        if not self._namespace_restrictions[index]:
+            return iso_c_binding[self._print(expr.dtype)][expr.precision]
+        self._macros[index] = True
+        return iso_c_binding_shortcut[self._print(expr.dtype)][expr.precision]
```

Comment:
`self._print(expr.dtype)` is called 5 times here! It would be better to call it once and save the result in a variable

Reply:
noted!